### PR TITLE
adds swagger YAML and JSON Task Management API files

### PR DESCRIPTION
**Pull request by imcoleman**

### CODE REVIEW

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 1]
```diff
@@ -0,0 +1,338 @@
...
+      tags:
+        - user
+      summary: Update an existing user
+      description: ''
+      operationId: updateUser
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: You need to add the {id} path parameter here too

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+    delete:
+      tags:
+        - user
+      summary: Deletes a user
+      description: ''
+      operationId: deleteUser
+      produces:
+        - application/json
+      parameters:
+        - name: api_key
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: api_key should be in every method if you want to add it but I will suggest you remove it completely from the Swagger

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+      tags:
+        - task
+      summary: Add a new task
+      description: ''
+      operationId: addTask
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: You need the user ID as parameter too

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+      tags:
+        - task
+      summary: Update an existing task
+      description: ''
+      operationId: updateTask
+      consumes:
+        - application/json
+      produces:
+        - application/json
+      parameters:
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: You need both IDs as parameters too

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+          description: Validation exception
+    delete:
+      tags:
+        - task
+      summary: Deletes a task
+      description: ''
+      operationId: deleteTask
+      produces:
+        - application/json
+      parameters:
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: You need both IDs as parameters and you can remove the api_key

File: fall-2017/icoleman/swagger/swagger.yaml [hunk 6]
```diff
@@ -0,0 +1,338 @@
...
+        enum:
+          - new
+          - in-progress
+          - blocked
+          - completed
+      owner:
+        type: integer
+        format: int64
+        description: the id for the user (that owns the task)
+  ArrayOfTasks:
```
Comment (toddysm) [inline fall-2017/icoleman/swagger/swagger.yaml]: This is wrong - you don't need separate object for array - it is built into the language

**Comment by imcoleman** [comment]
Thanks Toddy! Your comments were a big help, and helped me understand how
the code structure works.  I committed the changes to my repo, then closed
the original pull request and opened a new one (#27.)
-Ilsa

**Pull request closed without merging**